app/src/main/java/ca/bc/gov/bchealth/ui/healthrecord/TimeLineScreen.kt

```diff
@@ -0,0 +1,173 @@
+        modifier = modifier
+            .padding(start = 16.dp, top = 16.dp, end = 16.dp)
+            .fillMaxSize()
+    ) {
+
+        SearchBarUI(modifier = Modifier.fillMaxWidth(), onFilterClicked = onFilterClicked)
+        Spacer(modifier = Modifier.height(16.dp))
+        if (filtersApplied.isNotEmpty()) {
+            AppliedFilterUI(
+                modifier = Modifier.fillMaxWidth(),
+                filtersApplied,
+                onFilterCleared
+            )
+        }
+
+        if (healthRecords.isEmpty()) {
+            EmptyStateUI(
+                modifier,
+                image = R.drawable.ic_no_record,
+                title = R.string.no_records_found,
+                description = R.string.refresh
+            )
+        } else {
+            LazyColumn(
+                modifier
+                    .fillMaxSize(),
+                verticalArrangement = Arrangement.spacedBy(16.dp)
+            ) {
+
+                if (requiredProtectiveWordValidation) {
+                    item {
+                        HiddenMedicationRecordUI(onUnlockMedicationRecords = { onUnlockMedicationRecords() })
+                    }
+                }
+                items(healthRecords) { record ->
+                    HealthRecordItemUI(
+                        image = record.icon,
+                        title = record.title,
+                        description = record.description,
+                        onClick = { onClick(record) }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun HiddenMedicationRecordUI(onUnlockMedicationRecords: () -> Unit) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth(),
+        shape = MaterialTheme.shapes.medium,
+        elevation = 4.dp,
+    ) {
+
+        Column(
+            modifier = Modifier
+                .padding(16.dp)
+                .fillMaxWidth()
+        ) {
+            Text(
+                text = stringResource(id = R.string.hidden_medication_records),
+                style = MaterialTheme.typography.h3,
+                color = blue,
+                maxLines = 2,
+            )
+
+            Spacer(modifier = Modifier.height(8.dp))
+            Text(
+                text = stringResource(id = R.string.enter_protective_word_to_access_medication_records),
+                style = MaterialTheme.typography.smallButton,
+                color = grey,
+                fontWeight = FontWeight.Normal,
+                maxLines = 2
+            )
+
+            Spacer(modifier = Modifier.height(8.dp))
+
+            HGLargeButton(
+                modifier = Modifier.fillMaxWidth(),
+                onClick = { onUnlockMedicationRecords() },
+                text = stringResource(id = R.string.unlock_records),
+                leadingIcon = {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_lock),
+                        contentDescription = "lock"
```

Comment:
Check with Claire a meaningful description for this content. Maybe something like: "Unlock Record"?